command-expansion-server/src/app.ts

```diff
@@ -9,93 +9,89 @@ import { processDictionary } from "./packages/lib/CommandTypeCodegen.js";
 const PORT: number = parseInt(getEnv().PORT, 10) ?? 3000;
 
 const app: Application = express();
-app.use(bodyParser.json({ limit: '25mb' }));
+app.use(bodyParser.json({ limit: "25mb" }));
 
 DbExpansion.init();
 const db = DbExpansion.getDb();
 
-
 app.get("/", (req: Request, res: Response) => {
   res.send("Aerie Command Service");
 });
 
-app.put("/dictionary", async (req, res) => {
-  const dictionary = Buffer.from(req.body.input.dictionary, 'base64').toString();
-
-
+app.post("/dictionary", async (req, res) => {
+  const base64Dictionary: string = req.body.input.dictionary;
+  const dictionary = Buffer.from(base64Dictionary, "base64").toString("utf8");
   console.log(`Dictionary received`);
+
   const parsedDictionary = ampcs.parse(dictionary);
   console.log(
     `Dictionary parsed - version: ${parsedDictionary.header.version}, mission: ${parsedDictionary.header.mission_name}`
   );
+
   const commandDictionaryPath = await processDictionary(parsedDictionary);
   console.log(`command-lib generated - path: ${commandDictionaryPath}`);
 
   const sqlExpression = `
-    insert into command_dictionary (command_types, mission, version)
-    values ($1, $2, $3)
-    on conflict (mission, version) do update
-    set command_types = $1
-    returning id;
-  `;
+      insert into command_dictionary (command_types, mission, version)
+      values ($1, $2, $3)
+      on conflict (mission, version) do update
+      set command_types = $1
+      returning id;
+    `;
 
   const { rows } = await db.query(sqlExpression, [
     commandDictionaryPath,
     parsedDictionary.header.mission_name,
     parsedDictionary.header.version,
   ]);
 
-
-
   if (rows.length < 0) {
     console.error(`POST /dictionary: No command dictionary was updated in the database`);
-    res.status(500).send(`POST /dictionary: No command dictionary was updated in the database`);
-    return;
+    res.status(400).json({ message: `POST /dictionary: No command dictionary was updated in the database` });
```

Comment:
This commit, "Minor tweaks to Upload endpoint to comply with Hasura action", mixes style refactoring with functional changes, and as a result the meat of the change gets lost in the noise.

Would it be possible to split it into two commits? The first commit would only contain the style changes, and the second one would only change the functional changes?

Reply:
Hi @mattdailis the style change was a side-affect from the prettier javascript extension that runs every time I save. I agree this adds some noise, but what you highlighted is the change reflected in the commit. I had a lot of trouble rebasing and minimizing the PR into a smaller size? Can I pass on splitting this up into 2 commits?

Reply:
@goetzrrGit Can you do the rebase to split this apart?

Reply:
let @mattdailis and @dyst5422  let me try that

Reply:
Thanks for splitting the commits Ryan - I really appreciate it. (And sorry for missing your message from an hour ago)

Looks like this line was lost in the rebase:
```
res.status(400).json({ message: `POST /dictionary: No command dictionary was updated in the database` });
```

You'll probably want to add that back into the commit called "Don't return an 'undefined' if there is an error processing the command dictionary."

Reply:
Sigh....stupid git

Good caught. Thanks @mattdailis

Reply:
Fantastic, thanks Ryan